client/dali_grpc_client.py

```diff
@@ -120,44 +127,36 @@ def main():
 
     print("Loading images")
 
-    image_data = load_images(FLAGS.img_dir if FLAGS.img_dir is not None else FLAGS.img)
+    image_data = load_images(FLAGS.img_dir if FLAGS.img_dir is not None else FLAGS.img,
+                             max_images=FLAGS.batch_size * FLAGS.n_iter)
+
     image_data = array_from_list(image_data)
+    print("Images loaded")
 
-    print("Images loaded, inferring")
+    for batch in tqdm(utils.batcher(image_data, FLAGS.batch_size, n_iterations=FLAGS.n_iter),
+                      desc="Inferring", total=FLAGS.n_iter):
 
-    # Infer
-    inputs = []
-    outputs = []
-    input_name = "INPUT"
-    output_name = "OUTPUT"
-    input_shape = list(image_data.shape)
-    input_shape[0] = FLAGS.batch_size
-    inputs.append(tritonclient.grpc.InferInput(input_name, input_shape, "UINT8"))
-    outputs.append(tritonclient.grpc.InferRequestedOutput(output_name))
+        inputs, outputs = generate_io(FLAGS.input_name, FLAGS.output_name, batch.shape, "UINT8")
 
-    for batch in batcher(image_data, FLAGS.batch_size):
-        print("Input mean before backend processing:", np.mean(batch))
         # Initialize the data
         inputs[0].set_data_from_numpy(batch)
```

Comment:
Just a nitpick, but I would suggest packing that into
``` 
input = generate_input(FLAGS.input_name, batch) # you can acquire shape and type here.
output = generate_output(FLAGS.output_name)
```
better encapsulation, makes more sense